biz.aQute.bndlib/src/aQute/bnd/cdi/CDIAnnotations.java

```diff
@@ -292,7 +292,7 @@ private Discover findDiscoveryMode(Resource beansResource) {
 				try {
 					Object beanDiscoveryMode = discoveryModeExpression.evaluate(document, XPathConstants.STRING);
 
-					return Discover.valueOf(beanDiscoveryMode.toString());
+					return Discover.parse(beanDiscoveryMode.toString());
 				} catch (NullPointerException | XPathExpressionException e) {
```

Comment:
Also, you should catch `IllegalArgumentException` which can be thrown by `Enum.valueOf`, no? Since the XML could be corrupt and we should not explode.